Common/Data/Custom/USEnergyInformation.cs

```diff
@@ -0,0 +1,199 @@
+
+        /// <summary>
+        /// Returns true if the EIA API token has been set.
+        /// </summary>
+        public static bool IsAuthCodeSet { get; private set; }
+
+        /// <summary>
+        /// Sets the EIA API token.
+        /// </summary>
+        /// <param name="authCode">The EIA API token</param>
+        public static void SetAuthCode(string authCode)
+        {
+            if (string.IsNullOrWhiteSpace(authCode)) return;
+
+            AuthCode = authCode;
+            IsAuthCodeSet = true;
+        }
+
+        /// <summary>
+        /// Return the Subscription Data Source gained from the URL
+        /// </summary>
+        /// <param name="config">Configuration object</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>Subscription Data Source.</returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            return new SubscriptionDataSource(
+                $"https://api.eia.gov/series/?api_key={AuthCode}&series_id={config.Symbol}",
+                SubscriptionTransportMedium.Rest,
+                FileFormat.Collection);
+        }
+
+        /// <summary>
+        /// Reader converts each line of the data source into BaseData objects.
+        /// </summary>
+        /// <param name="config">Subscription data config setup object</param>
+        /// <param name="content">Content of the source document</param>
+        /// <param name="date">Date of the requested data</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        /// Collection of USEnergyInformation objects
+        /// </returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string content, DateTime date, bool isLiveMode)
+        {
+            // Do not emit if the content did not change
+            if (_previousContent == content) return null;
+            _previousContent = content;
+
+            try
+            {
+                var format = GetFormat(config.Symbol.Value);
+                var series = JObject.Parse(content)["series"][0];
+
+                // Do not emit if the end of the series did not change
+                date = DateTimeConverter(series["end"], format);
+                if (_previousDate == date) return null;
+                _previousDate = date;
+
+                var objectList = (
+                    from jToken in series["data"]
+                    where jToken[1].Type != JTokenType.Null
+                    select new USEnergyInformation
+                    {
+                        Symbol = config.Symbol,
+                        Period = Period,
+                        Value = (decimal)jToken[1],
+                        Time = DateTimeConverter(jToken[0], format)
```

Comment:
Given the long delays in emitting; please update to factor in an approximation:
`EndTime = jToken[0] + (lastUpdatedTime-endTime)`

Also given it is now EndTime focused please update Quarters to 0331 etc.